llvmlite/ir/instructions.py

```diff
@@ -162,7 +162,7 @@ def descr(self, buf):
 class PredictableInstr(Instruction):
 
     def set_weights(self, weights):
-        operands = [MetaDataString(self.module, "branch_weights")]
+        operands = [MetaDataString(self.module, b"branch_weights")]
```

Comment:
Hmm, I see. It seems there are two conflicting constraints here:
- `_escape_string`, given its current docstring ("Escape the given bytestring for safe use as a LLVM array constant"), should not accept unicode strings
- metadata strings definitely feel like "text" and it would be awkward to refuse Python 3 string literals for them (Numba has a couple examples of them and I'm sure other users would, too)

So perhaps `_escape_string` should be renamed to e.g. `_escape_bytes` (for use by array constants), and a wrapper function `_escape_string` created (for use by metadata strings). What do you think?